### PR TITLE
apollo server testing initialized

### PR DESCRIPTION
**Pull request by ilkkamaksy**
Closes issue #137

### CODE REVIEW

File: backend/src/tests/userschema.test.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      gitHubToken
+    }
+  }
+`
+
+describe('User schema mutations', () => {
+
+  it('user can not authorize with GitHub without providing a code as parameter', async () => {
+    const { mutate } = createTestClient(server)
+
```
Comment (guotin) [inline backend/src/tests/userschema.test.ts:42]: Tried to figure out what the typescript error for this line was but couldn't. Seems #153 just disabled similar error for the entire testfile. Maybe just add it here aswell if there's no other cure.

File: backend/src/tests/userschema.test.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    const res = await mutate({
+      mutation: AUTHORIZE_WITH_GH
+    })
+    
+    expect(res.errors?.some(error => error.message === 'Variable "$code" of required type "String!" was not provided.')).toBeTruthy()
```
Comment (guotin) [inline backend/src/tests/userschema.test.ts:47]: The assert (`expect`) could be easier to read if the response/error is first fetched to a variable and compared after

Comment (sinyman) [reply backend/src/tests/userschema.test.ts:47]: +1

File: backend/src/tests/userschema.test.ts (hunk 1 above)
Comment (ilkkamaksy) [reply backend/src/tests/userschema.test.ts:42]: Have you tried deleting node_modules and reinstalling?

File: backend/src/tests/userschema.test.ts (hunk 2 above)
Comment (ilkkamaksy) [reply backend/src/tests/userschema.test.ts]: Done.

File: backend/src/tests/userschema.test.ts (hunk 1 above)
Comment (tulma95) [reply backend/src/tests/userschema.test.ts:42]: `/* eslint-disable @typescript-eslint/unbound-method */` talking about this error?

Comment (guotin) [reply backend/src/tests/userschema.test.ts:42]: `Avoid referencing unbound methods which may cause unintentional scoping of `this`.eslint@typescript-eslint/unbound-method` persists even with new node_modules. It compiles but vscode mentions this as an issue.

**Review by guotin — APPROVED** [review]

**Review by tulma95 — APPROVED** [review]

**Pull request merged**